addons/web_editor/static/src/js/wysiwyg/MoveNodePlugin.js

```diff
@@ -0,0 +1,374 @@
+        node.parentElement.getAttribute('contentEditable') === 'true';
+}
+
+function isPointInside(rect, x, y) {
+    return rect.left <= x &&
+        rect.right >= x &&
+        rect.top <= y &&
+        rect.bottom >= y;
+};
+
+export class MoveNodePlugin {
+    constructor(options = {}) {
+        this._options = options;
+    }
+
+    start() {
+        this._editor = this._options.editor;
+        this._editable = this._options.editor.editable;
+        this._document = this._options.editor.document;
+
+        this._editor.addDomListener(this._editable, 'mousemove', this._onMousemove.bind(this), true);
+        this._editor.addDomListener(this._editor.document, 'keydown', this._onDocumentKeydown.bind(this), true);
+        this._editor.addDomListener(this._editor.document, 'mousemove', this._onDocumentMousemove.bind(this), true);
+
+        const positionOption = {
+            position: 'before',
+            targetContainerId: 'oe-avatars-overlap-container'
+        };
+        // This container help to add zone into which the mouse can activate the move widget.
+        this._widgetHookContainer = this._editor.makeAbsoluteContainer('oe-widget-hooks-container', positionOption);
+        // This container contains the differents widgets.
+        this._widgetContainer = this._editor.makeAbsoluteContainer('oe-widgets-container', positionOption);
+        // This container contains the jquery helper element.
+        this._dragHelperContainer = this._editor.makeAbsoluteContainer('oe-movenode-helper-container', positionOption);
+        // This container contains drop zones. They are the zones that handle where the drop should happen.
+        this._dropzonesContainer = this._editor.makeAbsoluteContainer('oe-dropzones-container', positionOption);
+        // This container contains drop hint. The final rectangle showed to the user.
+        this._dropzoneHintContainer = this._editor.makeAbsoluteContainer('oe-dropzone-hint-container', positionOption);
+
+        // Uncomment line for debugging tranparent zones
+        // this._widgetHookContainer.classList.add('debug');
+        // this._dropzonesContainer.classList.add('debug');
+
+        this._scrollableElement = this._editable.parentElement;
+        while (this._scrollableElement) {
+            if (getComputedStyle(this._scrollableElement).overflowY === 'auto') {
+                break;
+            }
+            this._scrollableElement = this._scrollableElement.parentElement;
+        }
+        this._scrollableElement = this._scrollableElement || this._editable;
+
+        this._updateHooks();
+        this.observer = new MutationObserver(() => {
+            this._removeMoveWidget && this._removeMoveWidget();
+            this._updateHooks();
+        });
+        this.observer.observe(this._editable, {
+            childList: true,
+            subtree: true,
+            characterData: true,
+            characterDataOldValue: true,
+        });
```

Comment:
I don't get it... you _observe_ every change in the editor for that ? Isn't that a bit overkill ? I mean you only see the handle on your current block anyway. I feel like I missing some important piece of understanding here.

Reply:
Perfs looks good on my machine. Between 2 and 3ms for `_updateHooks`.